### PR TITLE
Feature/update checkstyle codestyle

### PR DESCRIPTION
**Pull request by garyluu**
Updating the checkstyle and codestyle to match Google's

Had to override a small amount of things in the codestyle
Original: https://raw.githubusercontent.com/google/styleguide/gh-pages/intellij-java-google-style.xml
Had to override quite a bit more in the checkstyle (could probably override less but...oh well)
Original: https://raw.githubusercontent.com/checkstyle/checkstyle/checkstyle-8.34/src/main/resources/google_checks.xml

Search "Dockstore" in the checkstyle/codestyle to see what's different

Also added a bunch of checkstyle rules from the original which Google doesn't have (some rules useful, some odd)

### CODE REVIEW

**Comment by garyluu** [comment]
Funny...https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=7614033 reports something I purposely overrode in Google's checkstyle

File: checkstyle.xml [hunk 1]
```diff
@@ -1,337 +1,370 @@
...
-        <!-- </module> -->
-
-        <!-- Following interprets the header file as regular expressions. -->
-        <!-- <module name="RegexpHeader"/>                                -->
-
-
-        <!-- Checks for imports                              -->
-        <!-- See http://checkstyle.sf.net/config_import.html -->
-        <module name="AvoidStarImport"/>
-        <!-- ban new references to old commons lang and configuration -->
```
Comment (denis-yuen) [inline checkstyle.xml:213]: Carry over the ban on these packages (causes problems in newer versions of Java and old versions of libraries we've updated past)

File: checkstyle.xml [hunk 2]
```diff
@@ -1,337 +1,370 @@
...
-                      value="sun, org.apache.commons.lang, org.apache.commons.configuration, org.apache.commons.httpclient"/>
+        <module name="LeftCurly">
+            <property name="tokens"
+              value="ANNOTATION_DEF, CLASS_DEF, CTOR_DEF, ENUM_CONSTANT_DEF, ENUM_DEF,
+                    INTERFACE_DEF, LAMBDA, LITERAL_CASE, LITERAL_CATCH, LITERAL_DEFAULT,
+                    LITERAL_DO, LITERAL_ELSE, LITERAL_FINALLY, LITERAL_FOR, LITERAL_IF,
+                    LITERAL_SWITCH, LITERAL_SYNCHRONIZED, LITERAL_TRY, LITERAL_WHILE, METHOD_DEF,
+                    OBJBLOCK, STATIC_INIT"/>
         </module>
-        <module name="RedundantImport"/>
```
Comment (denis-yuen) [inline checkstyle.xml:218]: Ditto for redundant and unused imports

File: checkstyle.xml [hunk 3]
```diff
@@ -1,337 +1,370 @@
...
-        <!-- See http://checkstyle.sf.net/config_design.html -->
-        <!--
-            <module name="DesignForExtension"/>
-            <module name="VisibilityModifier"/>
-        -->
-        <module name="InterfaceIsType"/>
-        <module name="FinalClass"/>
-        <module name="HideUtilityClassConstructor"/>
-        <module name="InnerTypeLast"/>
-        <module name="MutableException"/>
```
Comment (denis-yuen) [inline checkstyle.xml:321]: These coding error checks should be carried over.

**Review by denis-yuen — CHANGES_REQUESTED** [review]
Carry over some of the things we've already locked-down

**Review by denis-yuen — APPROVED** [review]

File: dockstore-language-plugin-parent/src/main/java/io/dockstore/language/MinimalLanguageInterface.java [hunk 4]
```diff
@@ -69,7 +69,8 @@
     /**
      * When indexing, Dockstore will distinguish between extra files that hold things like extra code, tools, configuration (imported descriptors) and test parameter files (example parameter sets used to run a workflow)
      */
-    enum GenericFileType { IMPORTED_DESCRIPTOR, TEST_PARAMETER_FILE, CONTAINERFILE }
+    enum GenericFileType { IMPORTED_DESCRIPTOR, TEST_PARAMETER_FILE, CONTAINERFILE
+    }
```
Comment (coverbeck) [inline dockstore-language-plugin-parent/src/main/java/io/dockstore/language/MinimalLanguageInterface.java:73]: Interesting how they format that. I don't think I like it, but oh well.

**Review by coverbeck — APPROVED** [review]

File: dockstore-language-plugin-parent/src/main/java/io/dockstore/language/MinimalLanguageInterface.java (hunk 4 above)
Comment (garyluu) [reply dockstore-language-plugin-parent/src/main/java/io/dockstore/language/MinimalLanguageInterface.java:73]: I think it's supposed to be:
```
enum GenericFileType {
        IMPORTED_DESCRIPTOR, TEST_PARAMETER_FILE, CONTAINERFILE
    }
```
but checkstyle and codestyle doesn't seem to enforce that or autofix it in that way

**Review by NatalieEO — APPROVED** [review]

**Pull request merged**